core/storage/classifier/gae_models.py

```diff
@@ -181,6 +181,24 @@ def create(
         training_job_instance.put()
         return instance_id
 
+    @classmethod
+    def query_training_jobs(cls, cursor):
```

Comment:
May be make cursor a default argument and modify structure of `fetch_next_job` a bit so that you don't have to pass cursor when calling this function for first time. (Make first call outside while loop and then fetch more at the end of each loop, just before going to next iteration.)